STMicroelectronics/STM32L4_L4+/app/az_ulib_dm/src/az_ulib_dm_blob_ustream_interface.c

```diff
@@ -0,0 +1,319 @@
+  } while ((remain_size_local_buffer > 0) && ((result == AZ_OK) || (result == AZ_ULIB_EOF)));
+
+  // load total size copied into returned size if successful
+  if ((result == AZ_OK) || (result == AZ_ULIB_EOF))
+  {
+    *size = total_size_copied;
+    result = AZ_OK;
+  }
+
+  // set result to EOF if we have finished processing the last chunk
+  if (blob_http_cb->_internal.last_chunk && (remain_size_inner_buffer <= 0))
+  {
+    result = AZ_ULIB_EOF;
+  }
+
+  return result;
+}
+
+static az_result concrete_get_remaining_size(az_ulib_ustream* ustream_instance, size_t* const size)
+{
+  *size = ustream_instance->length - ustream_instance->inner_current_position;
+
+  return AZ_OK;
+}
+
+static az_result concrete_get_position(az_ulib_ustream* ustream_instance, offset_t* const position)
+{
+  return AZ_ERROR_NOT_SUPPORTED;
+}
+
+static az_result concrete_release(az_ulib_ustream* ustream_instance, offset_t position)
+{
+  return AZ_OK;
+}
+
+static az_result concrete_clone(
+    az_ulib_ustream* ustream_instance_clone,
+    az_ulib_ustream* ustream_instance,
+    offset_t offset)
+{
+  return AZ_ERROR_NOT_SUPPORTED;
+}
+
+static az_result concrete_dispose(az_ulib_ustream* ustream_instance)
+{
+  AZ_ULIB_TRY
+  {
+    az_ulib_ustream_data_cb* control_block = ustream_instance->control_block;
+    az_blob_http_cb* blob_http_cb = (az_blob_http_cb*)control_block->ptr;
+
+    // dispose of http client control block
+    AZ_ULIB_THROW_IF_AZ_ERROR(_az_nx_blob_client_dispose(
+        &blob_http_cb->_internal.http_client, &blob_http_cb->_internal.packet_ptr));
+
+    // dispose of the ustream control block
+    destroy_control_block(control_block);
+  }
+  AZ_ULIB_CATCH(...) {}
+
+  return AZ_ULIB_TRY_RESULT;
+}
+
+static az_result az_blob_ustream_init(
+    az_ulib_ustream* ustream_instance,
+    az_ulib_ustream_data_cb* ustream_data_cb,
+    az_ulib_release_callback ustream_data_cb_release_callback,
+    const uint8_t* const data_buffer,
+    size_t data_buffer_length,
+    az_ulib_release_callback blob_http_cb_release_callback)
+{
+  ustream_data_cb->api = &api;
+  ustream_data_cb->ptr = (const az_ulib_ustream_data*)data_buffer;
+  ustream_data_cb->data_release = blob_http_cb_release_callback;
+  ustream_data_cb->control_block_release = ustream_data_cb_release_callback;
+
+  init_instance(ustream_instance, ustream_data_cb, 0, 0, data_buffer_length);
+
+  return AZ_OK;
+}
+
+AZ_NODISCARD az_result az_blob_create_ustream_from_blob(
+    az_ulib_ustream* ustream_instance,
+    az_ulib_ustream_data_cb* ustream_data_cb,
+    az_ulib_release_callback ustream_data_cb_release_callback,
+    az_blob_http_cb* blob_http_cb,
+    az_ulib_release_callback blob_http_cb_release_callback,
+    NXD_ADDRESS* ip,
+    int8_t* resource,
+    int8_t* host)
+{
+  _az_PRECONDITION_NOT_NULL(ustream_instance);
+  _az_PRECONDITION_NOT_NULL(ustream_data_cb);
+  _az_PRECONDITION_NOT_NULL(blob_http_cb);
+  _az_PRECONDITION_NOT_NULL(ip);
+  _az_PRECONDITION(sizeof(resource) > 0);
```

Comment:
>sizeof(resource) [](http://example.com/codeflow?start=19&length=16)

`sizeof(resource)` is the size of the pointer in bytes, probably 4 for cortexM family. You need here is `strlen`. For `host` as well.